server/game/cards/08.6-SAT/ToGoForwardYouMustGoBack.js

```diff
@@ -0,0 +1,59 @@
+const _ = require('underscore');
+
+const DrawCard = require('../../drawcard.js');
+
+class ToGoForwardYouMustGoBack extends DrawCard {
+    setupCardAbilities() {
+        this.action({
+            title: 'Shuffle hand',
+            condition: () => _.any(this.game.getPlayers(), this.hasHand),
+            handler: () => {
+                this.remainingPlayers = this.game.getPlayersInFirstPlayerOrder();
+                this.selections = [];
+                this.proceedToNextStep();
+            }
+        });
+    }
+
+    hasHand(player) {
+        return player.hand.length > 0;
+    }
+
+    proceedToNextStep() {
+        if(this.remainingPlayers.length > 0) {
+            let currentPlayer = this.remainingPlayers.shift();
+            if(!this.hasHand(currentPlayer)) {
+                this.proceedToNextStep();
+                return true;
+            }
+
+            this.game.promptWithMenu(currentPlayer, this, {
+                activePrompt: {
+                    menuTitle: 'Shuffle your hand to draw 5 cards?',
+                    buttons: [
+                        { text: 'Yes', method: 'shuffleHand' },
+                        { text: 'No', method: 'cancel' }
+                    ]
+                },
+                source: this
+            });
+        }
+    }
+
+    shuffleHand(player) {
+        _.each(player.hand, card => player.moveCard(card, 'draw deck'));
```

Comment:
To remove the underscore usage, this can be rewritten as:

```javascript
for(let card of player.hand) {
    player.moveCard(card, 'draw deck');
}
```